tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Retargeting/ProjectRetargetHandlerTests.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.IO;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Retargeting;
+
+public class ProjectRetargetHandlerTests
+{
+    [Fact]
+    public async Task CheckForRetargetAsync_WhenNoValidOptions_ReturnsNull()
+    {
+        var handler = CreateInstance();
+
+        var result = await handler.CheckForRetargetAsync(RetargetCheckOptions.None);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task CheckForRetargetAsync_WhenRetargetingServiceIsNull_ReturnsNull()
+    {
+        var handler = CreateInstance(trackProjectRetargeting: null);
+
+        var result = await handler.CheckForRetargetAsync(RetargetCheckOptions.ProjectLoad);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task CheckForRetargetAsync_WhenNoGlobalJson_ReturnsNull()
+    {
+        var fileSystem = new IFileSystemMock();
+        var solution = CreateSolutionWithDirectory(@"C:\Solution");
+
+        var handler = CreateInstance(fileSystem: fileSystem, solution: solution);
+
+        var result = await handler.CheckForRetargetAsync(RetargetCheckOptions.ProjectLoad);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task CheckForRetargetAsync_WhenGlobalJsonHasNoSdkVersion_ReturnsNull()
+    {
+        var fileSystem = new IFileSystemMock();
+        var solution = CreateSolutionWithDirectory(@"C:\Solution");
+
+        // Create global.json without sdk.version
+        string globalJsonPath = @"C:\Solution\global.json";
+        await fileSystem.WriteAllTextAsync(globalJsonPath, "{}");
+
+        var handler = CreateInstance(fileSystem: fileSystem, solution: solution);
+
+        var result = await handler.CheckForRetargetAsync(RetargetCheckOptions.ProjectLoad);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task CheckForRetargetAsync_WhenNoRetargetVersionAvailable_ReturnsNull()
+    {
+        var fileSystem = new IFileSystemMock();
+        var solution = CreateSolutionWithDirectory(@"C:\Solution");
+
+        // Create global.json with sdk version
+        string globalJsonPath = @"C:\Solution\global.json";
+        await fileSystem.WriteAllTextAsync(globalJsonPath, @"{""sdk"":{""version"":""8.0.100""}}");
```

Comment:
nit: Raw strings (`"""`) can make JSON literals easier to read. I think VS will also syntax highlight them.